startup/userSetup.py

```diff
@@ -66,16 +66,79 @@ def bootstrap_sgtk():
         )
         return
 
+
+def load_sgtk_plugins():
+    """
+    Parse environment variables for a list of plugins to load that will
+    ultimately startup Toolkit and the tk-maya engine and environment.
+    """
+    for plugin_path in os.environ["SGTK_LOAD_MAYA_PLUGINS"].split(os.pathsep):
+        # Find the appropriate "plugin" sub directory. Maya will not be
+        # able to find any plugins under the base directory without this.
+        if os.path.isdir(os.path.join(plugin_path, "plug-ins")):
+            load_path = os.path.join(plugin_path, "plug-ins")
+        elif os.path.isdir(os.path.join(plugin_path, "plugins")):
+            load_path = os.path.join(plugin_path, "plugins")
+        else:
+            load_path = plugin_path
+
+        # Display a message indicating what path plugins are 
+        # being loaded from.
+        OpenMaya.MGlobal.displayInfo(
+            "Shotgun: Loading plugins from '%s'" % load_path
+        )
+
+        # Load the plugins from the resolved path
+        loaded_plugins = cmds.loadPlugin("%s/*" % load_path)
+        if not loaded_plugins:
+            OpenMaya.MGlobal.displayWarning(
+                "Shotgun: No plugins loaded from '%s'" % load_path
+            )
+            continue
+
+        # Set the newly loaded plugins to autoload.
+        for loaded_plugin in loaded_plugins:
+            cmds.pluginInfo(loaded_plugin, e=True, autoload=True)
```

Comment:
Should we auto-load them? What if the user launches maya without desktop next? Then the module path is not set and the plugin isn't loaded! Sorry if i am missing something :)